tests/test_types.py

```diff
@@ -46,18 +46,34 @@ def test_item_type():
     assert 'invalid choice: x' in str(e.value)
 
 
-@mock.patch('planet.scripts.types.get_asset_types',
-            new=mock.Mock(return_value=DEFAULT_ASSET_TYPES))
-def test_asset_type():
-    check = convert_asserter(AssetType())
-    check('visual*', ['visual', 'visual_xml'])
-    check('*data_*', ['metadata_aux', 'metadata_txt'])
-    check('analytic', ['analytic'])
-    check('analytic,visual', ['analytic', 'visual'])
+@mock.patch(
+    "planet.scripts.types.get_asset_types",
+    new=mock.Mock(return_value=DEFAULT_ASSET_TYPES),
+)
+@pytest.mark.parametrize(
+    "pattern,type_list",
+    [
+        ("visual*", ["visual", "visual_xml"]),
+        ("*data_*", ["metadata_aux", "metadata_txt"]),
+        ("analytic", ["analytic"]),
+        ("analytic,visual", ["analytic", "visual"]),
+    ],
+)
+def test_asset_type_convert(pattern, type_list):
+    """Patterns are correctly converted to a list of assert types."""
+    assert sorted(AssetType()(pattern)) == sorted(type_list)
```

Comment:
The code in convert_asserter is replaced by a single line. The assertion is in the body of the test, no need to check elsewhere to see what's going on.